src/main/java/uk/gov/justice/hmpps/casenotes/config/TokenVerifyingAuthManager.kt

```diff
@@ -0,0 +1,51 @@
+package uk.gov.justice.hmpps.casenotes.config
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpHeaders
+import org.springframework.security.authentication.AuthenticationManager
+import org.springframework.security.core.Authentication
+import org.springframework.security.oauth2.jwt.JwtDecoder
+import org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken
+import org.springframework.security.oauth2.server.resource.InvalidBearerTokenException
+import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationProvider
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+
+@Suppress("SpringJavaInjectionPointsAutowiringInspection")
+@Component
+class TokenVerifyingAuthManager(jwtDecoder: JwtDecoder,
+                                private val tokenVerificationApiWebClient: WebClient,
+                                @Value("\${tokenverification.enabled:false}") private val tokenVerificationEnabled: Boolean) :
+    AuthenticationManager {
+
+  private val jwtAuthenticationProvider = JwtAuthenticationProvider(jwtDecoder)
+
+  init {
+    jwtAuthenticationProvider.setJwtAuthenticationConverter(AuthAwareTokenConverter())
+  }
+
+  override fun authenticate(authentication: Authentication): Authentication {
+
+    if (tokenVerificationEnabled) {
+      val bearer = authentication as BearerTokenAuthenticationToken
+
+      // firstly check the jwt is still valid
+      val tokenDto = tokenVerificationApiWebClient.post().uri("/token/verify")
+          .header(HttpHeaders.AUTHORIZATION, "Bearer ${bearer.token}")
```

Comment:
have to manually add the header in here as is called before the `UserContextFilter` is setup